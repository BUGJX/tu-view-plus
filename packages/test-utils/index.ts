export { default as defineGetter } from './define-getter';
