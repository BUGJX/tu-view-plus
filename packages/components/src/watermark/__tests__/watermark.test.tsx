import { mount } from '@vue/test-utils';
import { describe, expect, test } from 'vitest';
import Watermark from '../src/watermark.vue';

const AXIOM = 'Tu view is good';

describe('Watermark.vue', () => {
  test('render test', () => {
    const wrapper = mount(() => <Watermark>{AXIOM}</Watermark>);

    expect(wrapper.classes()).toContain('tu-watermark');
    expect(wrapper.html()).toMatchSnapshot();
  });
  test('slots', () => {
    const wrapper = mount(() => <Watermark>{AXIOM}</Watermark>);

    expect(wrapper.text()).toContain(AXIOM);
  });
});
