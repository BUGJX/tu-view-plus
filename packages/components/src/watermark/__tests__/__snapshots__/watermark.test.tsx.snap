// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Watermark.vue > render test 1`] = `"<div class=\\"tu-watermark\\" style=\\"position: relative; overflow: hidden;\\">Tu view is good</div>"`;
